packages/kit/src/core/create_manifest_data/index.js

```diff
@@ -103,7 +103,10 @@ export default function create_manifest_data({ config, output, cwd = process.cwd
 				throw new Error(`Invalid route ${file} — brackets are unbalanced`);
 			}
 
-			if (/.+\[\.\.\.[^\]]+\]/.test(segment) || /\[\.\.\.[^\]]+\].+/.test(segment)) {
+			if (
+				/.+\[\.\.\.[^\]]+\]/.test(segment) ||
+				(/\[\.\.\.[^\]]+\].+/.test(segment) && !/\[\.\.\.[^\]]+\]\.json$/.test(segment))
```

Comment:
I doubt we want a hard-coded check for `[...rest].json.js`. `.json` is a convention and it can be anything else by the user. Also, have you confirm with this change that a SvelteKit app still routes normally? I have a hunch there are other places to fix as well.

With that said, I'm not sure what's the path forward, or maybe we should stick with the issue workaround for now. I'll see if the others has an opinion on this.

Reply:
It might be good to make it more flexible in the future.

But there are strong reasons to implement that fix:
- `.json` is the framework convention for endpoints
- `.json` is used in the example project
- because of that we would create unnecessary friction for new users coming to the framework